Objects/obmalloc.c

```diff
@@ -103,6 +109,60 @@ _PyMem_RawFree(void *ctx, void *ptr)
     free(ptr);
 }
 
+static int
+check_alignment(size_t alignment)
+{
+    /* alignment must be a power of 2, multiple of sizeof(void*)
+       and greater than 0 */
+    return (alignment > 0
+            && (alignment % sizeof(void *)) == 0
+            && (alignment & (alignment - 1)) == 0);
+}
+
+static void *
+_PyMem_RawAlignedAlloc(void *ctx, size_t alignment, size_t size)
+{
+    assert(check_alignment(alignment));
+    if (!check_alignment(alignment)) {
```

Comment:
I really think we should trust `posix_memalign`. :)

    The posix_memalign() function shall fail if:

    [EINVAL]
        The value of the alignment parameter is not a power of two multiple of sizeof( void *).
    [ENOMEM]
        There is insufficient memory available with the requested alignment.

Reply:
So I'd put it in the Windows path.

Reply:
Done.

"I really think we should trust posix_memalign."

I didn't want to rely on the **glibc**. Since you confirmed that EINVAL is part of the POSIX standard, I agree to rely on it.